reverse_file_buffered.py

```diff
@@ -0,0 +1,38 @@
+from file_io import Io, FileIo
+
+
+def reverse_file_in_memory(io: Io, name: str, buf_size: int):
+    """Reverse a content of the file `name` using no more than buf_size of additional memory"""
+    assert buf_size >= 2, "Minimum allowed buf_size is 2"
+
+    length = io.file_length(name)
+    chunk = buf_size//2
+    output_buff = bytearray(chunk*2)
+    mid = length//2
+    left_offset = 0
+    right_offset = length - left_offset - min(chunk, mid - left_offset)
+
+
+    while left_offset < right_offset or 0 < min(chunk, mid - left_offset) < chunk:
+        if length <= chunk:
```

Comment:
Note that `length` and `chunk` are never updated in the loop. Thus you either always go into this condition or never